perl/t/meson.build

```
@@ -0,0 +1,14 @@
+# Nix-Perl Tests
+#============================================================================
+
+
+# src
+#---------------------------------------------------
+
+nix_perl_tests = files(
+    'init.t')
```

Let's use two spaces, and arrays, like https://mesonbuild.com/Style-guide.html suggests
```suggestion
nix_perl_tests = files([
  'init.t',
])
```
(and also trailing commas)

---

The style guide is wrong and should be fixed.

At minimum the use of arrays is redundant and produces noise. I accept that two spaces vs four is a bikeshed and while I sort of wish two spaces would not be pushed I may be in the minority here, but the array thing is absolutely definitely redundant.

---

a posssible alternative would be to create a project centric style guide or even a linter config

---

afaik `files()` returns an array object so it shouldnt be required to pass an array in, itll only get squashed anyway

---

`muon fmt` defaults to 4 spaces :D 

positional arguments to functions which take variadic arguments get flattened. Pass two arguments in, it gets interpreted as an array of posargs with two elements. Pass one array in, it gets interpreted as an array of posargs with one array-element in, and the array gets flattened to two elements.

Pass three arguments, two of which are arrays and one of which is a string, and it gets interpreted as an array of posargs with 3 elements, two of which get unpacked and merged together + the third standalone string. End result: a single array of strings that contains the union of strings reachable in the posargs.

This transparently handles `files('foo.c', core_sources, extra_sources)`.

---

Yes, that looks like what I'd do.